src/Symfony/Component/Security/Core/Util/ClassUtils.php

```diff
@@ -54,12 +40,6 @@ private function __construct()
      */
     public static function getRealClass($object)
     {
-        $class = is_object($object) ? get_class($object) : $object;
-
-        if (false === $pos = strrpos($class, '\\'.self::MARKER.'\\')) {
-            return $class;
-        }
-
-        return substr($class, $pos + self::MARKER_LENGTH + 2);
+        return AclClassUtils::getRealClass($object);
```

Comment:
please no. We don't want to force anyone using ``security-core`` to get ``security-acl``, and the cyclic dependency it creates makes dependency resolution more complex.
I don't think this small logic is worth adding a dependency. You could keep the logic here, deprecate it and then duplicate it in the new component (the logic here is *already* a duplicate of the Doctrine Common logic to avoid the dependency on it just for that case).